src/inject/dynamic-theme/css-rules.ts

```diff
@@ -41,23 +42,33 @@ function isCSSVariable(property: string) {
     return property.startsWith('--') && !property.startsWith('--darkreader');
 }
 
+function getParentGroups(rule: CSSRule, stack: string[] = []): string[] {
+    const {parentRule} = rule;
+    if (!parentRule || !(parentRule instanceof CSSMediaRule)) {
+        return stack;
+    }
+    stack.push(parentRule.conditionText);
+    return getParentGroups(parentRule, stack);
+}
+
 export function getCSSVariables(rules: CSSRuleList) {
-    const variables = new Map<string, string>();
+    const variables = new Map<string, Variable>();
```

Comment:
I suggest you to use `Map<string, Map<string, Variable>>`. The first string is a selector, and the nested key string is a custom property (CSS variable name). This way you will not need to split by ";" that can be unpredictable (e.g. some attribute value selector), and will not produce a single rule for each variable, but instead many variables will be packed inside a single selector.